lib/rest/routes/users.js

```diff
@@ -1,12 +1,14 @@
-
 const express = require('express');
+const validateSchema = require('../middlewares/validateSchema');
 const usersSrv = require('../../services').user;
 const logger = require('../../logger').admin;
 
+const userSchema = 'http://express-gateway.io/models/users.json';
+
 module.exports = function () {
   const router = express.Router();
 
-  router.post('/', function (req, res, next) {
+  router.post('/', validateSchema(userSchema), function (req, res, next) {
```

Comment:
what if I say that this validation should be as part of service ? 

let say you want to define another admin API endpoint\CLI through extension 
that somehow will create a user. (Let say some import endpoint) 

you will have to replicate the validation logic 
if it is part of the service it will be centralized

Reply:
Validation is happening as well as part of the service. This validation actually is redundant; I put it here as well so if the payload is invalid we can skip the whole lookup in the database and all that stuff. Does that make sense?

Reply:
but this means you have duplicated logic. and you are creating user, so the service should like 
-- validate schema or fail
-- check for dupes or fail
...

so it should not be a db call at all, right ?

Reply:
That's correct. I think we can remove this layer. Let me do that right now.

Reply:
Removed in a8686d2

Thanks for noticing this, now that I think about it it really does not make any sense to have it there as well 👍